Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/RemoveCommentedCodeAnalyzer.cs

```diff
@@ -0,0 +1,57 @@
+﻿// © 2023 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Philips.CodeAnalysis.Common;
+
+namespace Philips.CodeAnalysis.MaintainabilityAnalyzers.Documentation
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class RemoveCommentedCodeAnalyzer : DiagnosticAnalyzer
+	{
+		private const string Title = @"Remove commented code";
+		private const string MessageFormat = @"Remove commented code on line {0}.";
+		private const string Description = @"Remove commented code";
+		private const string Category = Categories.Documentation;
+
+		private static readonly DiagnosticDescriptor Rule = new(Helper.ToDiagnosticId(DiagnosticIds.RemoveCommentedCode), Title, MessageFormat, Category, DiagnosticSeverity.Error, isEnabledByDefault: true, description: Description);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+		public override void Initialize(AnalysisContext context)
+		{
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+			context.EnableConcurrentExecution();
+			context.RegisterSyntaxNodeAction(Analyze, SyntaxKind.CompilationUnit);
```

Comment:
Shoudn't we just register on SingleLineCommentTrivia?

Reply:
Trivia are not SyntaxNodes, so you can't filter on them. Copyright Analyzer does the same.